### PR TITLE
Admin timetable

### PR DESCRIPTION
**Pull request by Anshumanshrivastava**

### CODE REVIEW

File: app/Http/Controllers/ImportTimetableController.php [hunk 1]
```diff
@@ -898,6 +900,44 @@ public function listTimetable ()
         return view('admin.timetable.index', compact('timetables', 'ar'));
     }
 
+    public function deleteTimetable($id)
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:903]: add Request $request as parameter

File: app/Http/Controllers/ImportTimetableController.php [hunk 2]
```diff
@@ -898,6 +900,44 @@ public function listTimetable ()
...
+         }
+         return redirect()->back()->with('success',"Deleted Successfully");
+    }
+
+    public function deleteAllTimetable(Request $request)
+    {
+        $class_name  = $request->txtSerachByClass;
+        $section_name = $request->txtSerachBySection;
+
+        $days = \DB::select ("SELECT t.class_id, t.subject_id,t.id
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:923]: the table name is class_timing  the use of this table is timeTable you can either use classTimings or timeTables as a variable name not days

File: app/Http/Controllers/ImportTimetableController.php [hunk 3]
```diff
@@ -898,6 +900,44 @@ public function listTimetable ()
...
+        FROM tbl_class_timings t
+        left join tbl_student_subjects s on s.id = t.subject_id
+        left join tbl_student_classes c on c.id = t.class_id
+        where c.class_name = ? and c.section_name=?", [$class_name , $section_name]);
+
+        foreach($days as $gt) {
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:929]: after "as" part of foreach should be singular of whatever is before the "as"

File: app/Http/Controllers/ImportTimetableController.php [hunk 4]
```diff
@@ -898,6 +900,44 @@ public function listTimetable ()
...
+         $getTime     = ClassTiming::where('from_timing', $classTiming->from_timing)->get();
+         foreach($getTime as $gt){
+         $gt->delete();
+         }
+         return redirect()->back()->with('success',"Deleted Successfully");
+    }
+
+    public function deleteAllTimetable(Request $request)
+    {
+        $class_name  = $request->txtSerachByClass;
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:920]: make all the variables in camelCase

File: app/Http/Controllers/ImportTimetableController.php [hunk 5]
```diff
@@ -898,6 +900,44 @@ public function listTimetable ()
...
+        where c.class_name = ? and c.section_name=?", [$class_name , $section_name]);
+
+        foreach($days as $gt) {
+        $token = CommonHelper::varify_Admintoken(); // verify admin token           
+        if($gt!= ''){
+        $inv_delete = CommonHelper::teacher_invitation_delete($token,$gt->id); 
+        }
+
+        $getTime = ClassTiming::where('class_id',  $gt->class_id)->where("subject_id",$gt->subject_id);
+         $getTime->delete();
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:936]: alignment

File: app/Http/Controllers/ImportTimetableController.php [hunk 6]
```diff
@@ -898,6 +900,44 @@ public function listTimetable ()
         return view('admin.timetable.index', compact('timetables', 'ar'));
     }
 
+    public function deleteTimetable($id)
+    {
+        $timetableId = $id;
+        $token = CommonHelper::varify_Admintoken(); // verify admin token               
+        if($timetableId != ''){
+            $inv_delete = CommonHelper::teacher_invitation_delete($token,$timetableId);
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:908]: camelCase

File: routes/web.php [hunk 7]
```diff
@@ -89,6 +89,8 @@
     Route::post('/edit-timetable', 'ImportTimetableController@editTimetable')->name('timetable.edit');
     Route::match(array('GET', 'POST'), '/timetable-import', 'ImportTimetableController@timeTableImport')->name('admin.timetableimport');
     Route::get('/download-sample', 'ImportTimetableController@sampleDownload')->name('admin.sampleDownload');
+    Route::get('delete-timetable/{id}', 'ImportTimetableController@deleteTimetable')->name('delete-timetable');
```
Comment (krtrthbilory) [inline routes/web.php:92]: route should be like timetable/delete and timetable/deleteAll

File: app/Http/Controllers/ImportTimetableController.php [hunk 8]
```diff
@@ -898,6 +900,43 @@ public function listTimetable ()
         return view('admin.timetable.index', compact('timetables', 'ar'));
     }
 
+    public function deleteTimetable(Request $request,$id)
+    {
+        $timeTableId = $id;
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:905]: you can remove this line and use id anyway this is timetable controller and this is delete timetable itself

Comment (krtrthbilory) [reply app/Http/Controllers/ImportTimetableController.php:905]: what does this di consist

File: app/Http/Controllers/ImportTimetableController.php [hunk 9]
```diff
@@ -898,6 +900,53 @@ public function listTimetable ()
         return view('admin.timetable.index', compact('timetables', 'ar'));
     }
 
+    public function deleteTimetable(Request $request,$id)
+    {
+        $classTiming = ClassTiming::where('id', $id)->get()->first();
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:905]: try ClassTiming::find('id', $id); here

File: app/Http/Controllers/ImportTimetableController.php [hunk 10]
```diff
@@ -898,6 +900,53 @@ public function listTimetable ()
...
+        $classID     = $classTiming->class_id;
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:906]: no need to assign you can directly use these $classTiming->class_id inside eloquent

File: app/Http/Controllers/ImportTimetableController.php [hunk 11]
```diff
@@ -898,6 +900,53 @@ public function listTimetable ()
...
+        $token = CommonHelper::varify_Admintoken(); // verify admin token
+         if ($objTeacher) {
+            $gCode = $objTeacher->g_code;
+            if ( $gCode != '' ) {
+            $invDelete = CommonHelper::teacher_invitation_delete($token, $gCode); 
+             }
+           }
+         $timeTables = ClassTiming::where('from_timing', $classTiming->from_timing)->get();
+         foreach($timeTables as $timeTable){
+         $timeTable->delete();
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:919]: use correct indentation

File: app/Http/Controllers/ImportTimetableController.php [hunk 12]
```diff
@@ -898,6 +900,53 @@ public function listTimetable ()
...
+        $classTiming = ClassTiming::where('id', $id)->get()->first();
+        $classID     = $classTiming->class_id;
+        $teacherId   = $classTiming->teacher_id;
+        $subjectId   = $classTiming->subject_id;  
+        $objTeacher  = InvitationClass::where('class_id', $classID)->where('subject_id', $teacherId)->where('teacher_id', $subjectId)->get()->first();
+        $token = CommonHelper::varify_Admintoken(); // verify admin token
+         if ($objTeacher) {
+            $gCode = $objTeacher->g_code;
+            if ( $gCode != '' ) {
+            $invDelete = CommonHelper::teacher_invitation_delete($token, $gCode);
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:914]: correct indentation

File: app/Http/Controllers/ImportTimetableController.php [hunk 13]
```diff
@@ -898,6 +900,53 @@ public function listTimetable ()
...
+    public function deleteAllTimetable(Request $request)
+    {
+        $className  = $request->txtSerachByClass;
+        $sectionName = $request->txtSerachBySection;
+        
+        $timeTables = \DB::select ("SELECT t.class_id, t.subject_id,t.id,t.teacher_id
+        FROM tbl_class_timings t
+        left join tbl_student_subjects s on s.id = t.subject_id
+        left join tbl_student_classes c on c.id = t.class_id
+        where c.class_name = ? and c.section_name=?", [$className , $sectionName]);
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:933]: you can directly use $request->txtSearchByClass here no need to assign it to a variable

Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:933]: why this ?

Comment (Anshumanshrivastava) [reply app/Http/Controllers/ImportTimetableController.php:933]: Fetching timetable on the basis of class name and section then applying delete all on that

File: app/Http/Controllers/ImportTimetableController.php [hunk 14]
```diff
@@ -898,6 +900,47 @@ public function listTimetable ()
...
+    public function deleteAllTimetable(Request $request)
+    {
+        $timeTables = \DB::select ("SELECT t.class_id, t.subject_id,t.id,t.teacher_id
+        FROM tbl_class_timings t
+        left join tbl_student_subjects s on s.id = t.subject_id
+        left join tbl_student_classes c on c.id = t.class_id
+        where c.class_name = ? and c.section_name=?", [$request->txtSerachByClass,$request->txtSerachBySection]);
+
+        foreach($timeTables as $timeTable){
+        $objTeacher  = InvitationClass::where('class_id', $timeTable->class_id)->where('subject_id', $timeTable->subject_id)->where('teacher_id', $timeTable->teacher_id)->get()->first();
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:930]: indenntation

File: app/Http/Controllers/ImportTimetableController.php [hunk 15]
```diff
@@ -898,6 +900,47 @@ public function listTimetable ()
...
+        $token = CommonHelper::varify_Admintoken(); // verify admin token
+        if ($objTeacher){
+        $gCode = $objTeacher->g_code;
```
Comment (krtrthbilory) [inline app/Http/Controllers/ImportTimetableController.php:933]: idenntation

**Pull request merged**